src/docfx/lib/JsonUtility.cs

```diff
@@ -267,14 +267,11 @@ private static string BuildPath(string path, Type type)
             return path is null ? type.Name : $"{path}.{type.Name}";
         }
 
-        private static bool TypeContainsJsonExtensionData(Type type)
+        private static bool CheckTypeContainsJsonExtensionData(Type type)
         {
-            if (!s_cache.TryGetValue(type, out var containsJsonExtensionData))
-            {
-                containsJsonExtensionData = type.GetProperties().Any(prop => prop.GetCustomAttribute<JsonExtensionDataAttribute>() != null);
-                s_cache.TryAdd(type, containsJsonExtensionData);
-            }
-            return containsJsonExtensionData;
+            return s_cacheTypeContainsJsonExtensionData.GetOrAdd(
+                type,
+                new Lazy<bool>(() => type.GetProperties().Any(prop => prop.GetCustomAttribute<JsonExtensionDataAttribute>() != null))).Value;
```

Comment:
> Lazy<bool> [](start = 19, length = 11)

no need lazy?

Reply:
i know, but it's ok to two twice as it's stateless

Reply:
I think we need Lazy here to avoid multiple expensive instantiation of the value with the same key
[https://stackoverflow.com/questions/47489211/concurrentdictionary-lazy-would-instantiation-happen-only-once](https://stackoverflow.com/questions/47489211/concurrentdictionary-lazy-would-instantiation-happen-only-once)

---
In reply to: [207413221](https://github.com/dotnet/docfx/pull/3148#discussion_r207413221) [](ancestors = 207413221)

Reply:
we know that, just wandering whether the getting attribute is worth it